### PR TITLE
Add docgen documentation.

### PR DESCRIPTION
**Pull request by csyonghe**

### CODE REVIEW

**Review by jkwak-work — APPROVED** [review]
Looks good to me

File: docs/stdlib-docgen.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+Stdlib documentation is done directly in comments inside `source/slang/*.meta.slang` files.
+A documentation comment should be placed directly above the declaration, either insde a `/**   */` comment block, or
+after `///`. The following directives are allowed in comments:
+
+- `@param paramName description` documents a parameter or a generic parameter.
+- `@remarks` starts the remarks section.
+- `@see` starts the "See also" section.
+- `@return` starts the `Return value" section.
+- `@example` starts the "Example" section.
+- `@category categoryID Category Name` marks the decl to be in a category. The cateogry name is only required for the first time `categoryID` is used, and omitted for the remaining `@category` lines.
```
Comment (cheneym2) [inline docs/stdlib-docgen.md:48]: Typo: cateogry

File: docs/stdlib-docgen.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        ...
+    }
+```
+
+Note that unlike doxygen, the directives marks the start of a new section, and applies to all following paragraphs. You don't need to repetitively mark new paragraphs
+as with `@remarks`.
+
+## What to document
+
+- Provide a brief description of the declaration in under three sentenses.
```
Comment (cheneym2) [inline docs/stdlib-docgen.md:94]: Typo: sentenses

File: docs/stdlib-docgen.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+- Document all nuances, including target specific behaviors in the remarks section.
+- Include examples if needed in the examples section.
+- Provide a see also section with links to related declarations.
+
+After updating comments, build `slangc`, and run `slangc -compile-stdlib -doc` in `stdlib-reference` diretory to update the markdown files for preview.
```
Comment (cheneym2) [inline docs/stdlib-docgen.md:99]: Typo: diretory

**Pull request merged**